src/main.c

```diff
@@ -233,27 +402,113 @@ static void write_spc(FILE *f) {
 		BYTE *new_spc = memcpy(malloc(spc_size), res_data, spc_size);
 
 		// Copy packs/blocks to byte array
-		for (int pack = 0; pack < 3; pack++) {
-			pack_to_spc(packs_loaded[pack], new_spc);
+//		for (int pack = 0; pack < 3; pack++) {
+//			pack_to_spc(packs_loaded[pack], new_spc);
+//		}
+
+		// compile_song corrupts the spc and any potential samples/instruments, so we need to make a copy first...
+		BYTE spc_copy[0x10000];
+		memcpy(spc_copy, spc, 0x10000);
+		WORD *brr_copy = (WORD *)&spc_copy[sample_ptr_base];
+
+		// Move the music from wherever it was in the spc to 0x3200... (ripping out part of the program block in the process...)
+		const WORD dstMusic = 0x3200;
+		int size = compile_song(&cur_song);
+		memcpy(spc + dstMusic, &spc[cur_song.address], size);
+
+		// recompile so the addresses are correct...
+		cur_song.address = dstMusic;
+		compile_song(&cur_song);
+
+		const unsigned int NUM_INSTRUMENTS = 128;
+		const WORD dstSamplePointers = dstMusic + 0x100 + (size & 0xFF00);
+		const WORD dstInstruments = dstSamplePointers + 0x2*NUM_INSTRUMENTS;
+		const WORD inst_size = NUM_INSTRUMENTS*6;
+		const WORD dstSamples = dstInstruments + inst_size;
```

Comment:
I think NUM_INSTRUMENTS should be 64. This matches with Earthbound's memory layout:
```
6C00-6DFF: Sample table: struct { WORD start; WORD loop; } [64] // Okay actually this doesn't match, only takes up 0x100 bytes - so you could maybe move it closer and get more space!
6E00-6F7F: Instrument table: byte [6*64] // 6*64=0x180 bytes, the exact size normally allocated
6F80-6F97: Misc data tables, 0x18 bytes
```
Notably, having 128 instruments takes up 0x300 bytes for the instrument table, which is 2x the normal space, which is what leads me to say that NUM_INSTRUMENTS should be 64. Have you seen games with more than 64 instruments?

Reply:
Space Armada from StarFox uses 74 apparently unless something else is wrong somewhere. I've bumped it down to 80 and that seems to work.